src/adapter/graphql/src/root.rs

```diff
@@ -39,6 +39,7 @@ impl Query {
     ///
     /// Datasets are historical streams of events recorded under a certain
     /// schema.
+    #[graphql(guard = "AuthPolicyGuard")]
```

Comment:
A guard can be set on the struct level. Maybe not a good idea here, but try this in other places to avoid copying guard for each method.

![image](https://github.com/user-attachments/assets/3ded7bd0-8d34-48a5-b65c-7a78cf203239)